### PR TITLE
[Hexagon] Propagate QNN Concat Quantization Params to Inputs

### PR DESCRIPTION
**Pull request by arangasa**
This helps in using optimized implementations of concat; this PR also avoids redundant requantization when an input to a concat is already a requantize op.

### CODE REVIEW

**Comment by arangasa** [comment]
cc @kparzysz-quic @quic-sanirudh

File: python/tvm/contrib/hexagon/transform.py [hunk 1]
```diff
@@ -311,3 +318,73 @@ def remove_empty_pad(mod):
...
     mod["main"] = rewrite(remove_empty_pad_callback(), mod["main"])
     return mod
+
+
+class simplify_qnn_concat_in_func(DFPatternCallback):
+
+    """
+    Propagate qnn.concat's quantization params to its inputs,
+    and try to avoid redundant requantization while doing so.
+    """
```
Comment (quic-sanirudh) [inline python/tvm/contrib/hexagon/transform.py:328]: Minor comment. If possible we could add a snippet of the relay IR before and after this pass in the comments similar to the comments mentioned in the functions above (like `remove_empty_pad_callback` or `qdistilbert_rewrite`).

**Review by quic-sanirudh — APPROVED** [review]
LGTM apart from a minor comment.

**Comment by quic-sanirudh** [comment]
@tvm-bot rerun

**Pull request merged**